src/androidTest/java/com/mixpanel/android/mpmetrics/MixpanelNotificationBuilderTest.java

```diff
@@ -108,6 +119,32 @@ public void testInvalidIcon() {
         verify(builderSpy).setSmallIcon(DEFAULT_ICON_ID);
     }
 
+    public void testExpandedImageUsingValidUrl() throws ImageStore.CantGetImageException{
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        intent.putExtra("mp_imgurl", VALID_URL);
+
+        Bitmap fakeBitmap = getFakeBitmap();
+
+        when(mpPushSpy.getBitmapFromUrl(VALID_URL)).thenReturn(fakeBitmap);
+
+        mpPushSpy.createNotification(intent);
+        verify(mpPushSpy).getBitmapFromUrl(VALID_URL);
+        verify(builderSpy).setStyle(new Notification.BigPictureStyle().bigPicture(fakeBitmap));
+    }
+
+    public void testExpandedImageUsingInvalidUrl() throws ImageStore.CantGetImageException{
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        intent.putExtra("mp_imgurl", "");
+
+        when(mpPushSpy.getBitmapFromUrl(VALID_URL)).thenReturn(null);
```

Comment:
`mpPushSpy.getBitmapFromUrl(VALID_URL)` will never be called because you're sending an empty string to `mp_imgurl`. I don't think you need to mock this method because it shouldn't actually be called at all.

Reply:
Changed these tests